### PR TITLE
Added image processor class

### PR DESCRIPTION
**Pull request by mahimairaja**
# What does this PR do?

Fixes           https://github.com/huggingface/transformers/issues/26360

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)?
https://github.com/huggingface/transformers/issues/26360
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by mahimairaja** [comment]
Hi @shauray8! Kindly review the PR...

**Comment by shauray8** [comment]
Hi @mahimairaja I was going through the preprocessor, I think you should follow the preprocessor from *yolos* that will give you a better idea of how to structure it, without the proper design it wont pass all the tests

**Comment by shauray8** [comment]
@mahimairaja make this follow the *yolos* format, I'll merge this and make further changes myself

**Comment by mahimairaja** [comment]
Let me convert it into yolo format [WIP]

**Comment by mahimairaja** [comment]
Refactored... Kindly review now

**Comment by shauray8** [comment]
You should also add the post processor to it - https://github.com/sczhou/ProPainter/blob/main/inference_propainter.py#L454C1-L472C5

**Comment by mahimairaja** [comment]
Okay, I will add up post processing shortly...

**Review by mahimairaja — COMMENTED** [review]
Added image post processing

**Comment by shauray8** [comment]
apart from the get_device it looks good, there's some documentation to do but I would prefer you fix the device issue and I'll merge it and test all of it and make it work from there

**Comment by shauray8** [comment]
can we also return this https://github.com/sczhou/ProPainter/blob/main/inference_propainter.py#L262

**Comment by mahimairaja** [comment]
> can we also return this https://github.com/sczhou/ProPainter/blob/main/inference_propainter.py#L262

It is the integer version of the frames, probably we can convert it inside the modelling classes, else it may increase the bandwidth?

File: src/transformers/models/propainter/image_processing_propainter.py [hunk 1]
```diff
@@ -14,254 +14,425 @@
...
+
+        if self.do_resize:
+            frames, size, out_size = self.resize_frames(frames, size)
+
+        if self.mode == 'video_inpainting':
+            frames_len = len(frames)
+            flow_masks, masks_dilated = self.read_mask(self.mask, frames_len, size,
+                                                flow_mask_dilates=self.mask_dilation,
+                                                mask_dilates=self.mask_dilation)
+        elif self.mode == 'video_outpainting':
```
Comment (shauray8) [inline src/transformers/models/propainter/image_processing_propainter.py:301]: can you make 2 different functions one for inpainting and the other for outpainting as we have different classes in modelling for both of them?

File: src/transformers/models/propainter/image_processing_propainter.py [hunk 2]
```diff
@@ -14,254 +14,425 @@
...
-                for image in images
-            ]
-
-        images = [
-            to_channel_dimension_format(image, data_format, input_channel_dim=input_data_format) for image in images
-        ]
-
-        data = {"pixel_values": images}
+
+        device = self.get_device()
```
Comment (shauray8) [inline src/transformers/models/propainter/image_processing_propainter.py:285]: not needed

**Comment by mahimairaja** [comment]
> apart from the get_device it looks good, there's some documentation to do but I would prefer you fix the device issue and I'll merge it and test all of it and make it work from there

### Get Device is working

get_device


I tried explicitly to execute the `get_device()` function and it works as expected I hope...

**Review by shauray8 — CHANGES_REQUESTED** [review]
please address all the issues before merging, if there's anything you are confused about do tell me I'll handle it.

**Comment by shauray8** [comment]
> > apart from the get_device it looks good, there's some documentation to do but I would prefer you fix the device issue and I'll merge it and test all of it and make it work from there
> 
> ### Get Device is working
> get_device
> 
> I tried explicitly to execute the `get_device()` function and it works as expected I hope...

while inferencing the user maps it to the device themselves so we don't have to worry about that,  (we don't need to have  function to map to device)

**Comment by shauray8** [comment]
> > can we also return this https://github.com/sczhou/ProPainter/blob/main/inference_propainter.py#L262
> 
> It is the integer version of the frames, probably we can convert it inside the modelling classes, else it may increase the bandwidth?

Okay, maybe let's convert it into the modelling itself.

**Comment by mahimairaja** [comment]
Gotcha! Solving it ASAP

File: src/transformers/models/propainter/image_processing_propainter.py [hunk 3]
```diff
@@ -14,24 +14,20 @@
...
-    valid_images,
-)
+import os
+import re
+import torchvision
+import torch
+from torchvision import transforms
+from PIL import Image
+import cv2
+import scipy
```
[the remark this answers is not in the document]
Comment (mahimairaja) [reply src/transformers/models/propainter/image_processing_propainter.py]: Done!

File: src/transformers/models/propainter/image_processing_propainter.py [hunk 4]
```diff
@@ -14,24 +14,20 @@
...
-    is_scaled_image,
-    make_list_of_images,
-    to_numpy_array,
-    valid_images,
-)
+import os
+import re
+import torchvision
+import torch
+from torchvision import transforms
```
[the remark this answers is not in the document]
Comment (mahimairaja) [reply src/transformers/models/propainter/image_processing_propainter.py]: Done!

File: src/transformers/models/propainter/image_processing_propainter.py (hunk 1 above)
Comment (mahimairaja) [reply src/transformers/models/propainter/image_processing_propainter.py:301]: Will do it...

File: src/transformers/models/propainter/image_processing_propainter.py (hunk 2 above)
Comment (mahimairaja) [reply src/transformers/models/propainter/image_processing_propainter.py:285]: Okay will remove the device info`s

**Review by mahimairaja — COMMENTED** [review]
I resolved all the listed issues, kindly review them...

**Review by shauray8 — APPROVED** [review]
Looks good, thank you!

**Comment by mahimairaja** [comment]
Thank you! Happy to be a part...

**Comment by mahimairaja** [comment]
Kindly merge

**Pull request merged**